repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,664 @@
+                    elif option[0] == "burst":
+                        conf += " presend 6"
+                    elif option[0] == "true":
+                        conf += " trust"
+                    else:
+                        assert False
+                conf += "\n"
+        conf += "\n"
+
+        return conf
+
+    def get_chrony_conf_allows(self):
+        allowed_networks = filter(lambda n: "ignore" not in self.restrictions[n] and
+                                    "noserve" not in self.restrictions[n],
+                                  self.restrictions.keys())
+
+        conf = ""
+        for network in sorted(allowed_networks, key=lambda n: (n.version, n)):
+            if network.num_addresses > 1:
+                conf += "allow {}\n".format(network)
+            else:
+                conf += "allow {}\n".format(network.network_address)
+
+        if conf:
+            conf = "# Allow NTP client access.\n" + conf
+            conf += "\n"
+
+        return conf
+
+    def get_chrony_conf_cmdallows(self):
+        allowed_networks = filter(lambda n: "ignore" not in self.restrictions[n] and
+                                    "noquery" not in self.restrictions[n] and
+                                    n != ipaddress.ip_network(u"127.0.0.1/32") and
+                                    n != ipaddress.ip_network(u"::1/128"),
+                                  self.restrictions.keys())
+
+        ip_versions = set()
+        conf = ""
+        for network in sorted(allowed_networks, key=lambda n: (n.version, n)):
+            ip_versions.add(network.version)
+            if network.num_addresses > 1:
+                conf += "cmdallow {}\n".format(network)
+            else:
+                conf += "cmdallow {}\n".format(network.network_address)
+
+        if conf:
+            conf = "# Allow remote monitoring.\n" + conf
+            if 4 in ip_versions:
+                conf += "bindcmdaddress 0.0.0.0\n"
+            if 6 in ip_versions:
+                conf += "bindcmdaddress ::\n"
+            conf += "\n"
+
+        return conf
+
+    def get_chrony_conf(self, chrony_keys_path):
+        local_stratum = 0
+        maxdistance = 0.0
+        minsources = 1
+        orphan_stratum = 0
+        logs = []
+
+        for source in self.time_sources:
+            address = source["address"]
+            if address.startswith("127.127.1."):
+                if address in self.fudges and "stratum" in self.fudges[address]:
+                    local_stratum = int(self.fudges[address]["stratum"])
+                else:
+                    local_stratum = 5
+
+        for tos in self.tos_options:
+            if tos[0] == "maxdist":
+                maxdistance = float(tos[1])
+            elif tos[0] == "minsane":
+                minsources = int(tos[1])
+            elif tos[0] == "orphan":
+                orphan_stratum = int(tos[1])
+            else:
+                assert False
```

Comment:
assert False?

Reply:
That would fail is there was a unexpected tos (i.e. bug). Anyway, I've reworked this for the other issue with int() failing.